src/renderers/dom/shared/__tests__/DOMPropertyOperations-test.js

```diff
@@ -83,181 +64,105 @@ describe('DOMPropertyOperations', () => {
-        'allowFullScreen',
-        false,
-      );
-      expect(stubNode.hasAttribute('allowFullScreen')).toBe(false);
+      var container = document.createElement('div');
+      ReactDOM.render(<div allowFullScreen={false} />, container);
+      expect(container.firstChild.hasAttribute('allowFullScreen')).toBe(false);
     });
 
     it('should remove when setting custom attr to null', () => {
-      DOMPropertyOperations.setValueForProperty(stubNode, 'data-foo', 'bar');
-      expect(stubNode.hasAttribute('data-foo')).toBe(true);
-      DOMPropertyOperations.setValueForProperty(stubNode, 'data-foo', null);
-      expect(stubNode.hasAttribute('data-foo')).toBe(false);
-    });
-
-    it('should use mutation method where applicable', () => {
-      var foobarSetter = jest.fn();
-      // inject foobar DOM property
-      DOMProperty.injection.injectDOMPropertyConfig({
-        Properties: {foobar: null},
-        DOMMutationMethods: {
-          foobar: foobarSetter,
-        },
-      });
-
-      DOMPropertyOperations.setValueForProperty(
-        stubNode,
-        'foobar',
-        'cows say moo',
-      );
-
-      expect(foobarSetter.mock.calls.length).toBe(1);
-      expect(foobarSetter.mock.calls[0][0]).toBe(stubNode);
-      expect(foobarSetter.mock.calls[0][1]).toBe('cows say moo');
+      var container = document.createElement('div');
+      ReactDOM.render(<div data-foo="bar" />, container);
+      expect(container.firstChild.hasAttribute('data-foo')).toBe(true);
+      ReactDOM.render(<div data-foo={null} />, container);
+      expect(container.firstChild.hasAttribute('data-foo')).toBe(false);
     });
 
     it('should set className to empty string instead of null', () => {
-      DOMPropertyOperations.setValueForProperty(
-        stubNode,
-        'className',
-        'selected',
-      );
-      expect(stubNode.className).toBe('selected');
-
-      DOMPropertyOperations.setValueForProperty(stubNode, 'className', null);
+      var container = document.createElement('div');
+      ReactDOM.render(<div className="selected" />, container);
+      expect(container.firstChild.className).toBe('selected');
+      ReactDOM.render(<div className={null} />, container);
       // className should be '', not 'null' or null (which becomes 'null' in
       // some browsers)
-      expect(stubNode.className).toBe('');
-      expect(stubNode.getAttribute('class')).toBe(null);
+      expect(container.firstChild.className).toBe('');
+      expect(container.firstChild.getAttribute('class')).toBe(null);
     });
 
     it('should remove property properly for boolean properties', () => {
-      DOMPropertyOperations.setValueForProperty(stubNode, 'hidden', true);
-      expect(stubNode.hasAttribute('hidden')).toBe(true);
-
-      DOMPropertyOperations.setValueForProperty(stubNode, 'hidden', false);
-      expect(stubNode.hasAttribute('hidden')).toBe(false);
-    });
-
-    it('should remove property properly even with different name', () => {
```

Comment:
I removed this test. I couldn’t find any cases where property name would differ in practice—neither in master nor internally at FB. If we add such case we should add an integration test for that specific property instead.